wlauto/workloads/octane2/uiauto/src/org/linaro/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,93 @@
+/*    Copyright 2016 Linaro Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+
+package org.linaro.wlauto.uiauto.octane2;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.util.Log;
+
+// Import the uiautomator libraries
+import com.android.uiautomator.core.UiObject;
+import com.android.uiautomator.core.UiObjectNotFoundException;
+import com.android.uiautomator.core.UiSelector;
+
+import com.arm.wlauto.uiauto.BaseUiAutomation;
+
+public class UiAutomation extends BaseUiAutomation {
+
+    public static String TAG = "octane2";
+    public String[] categories = {
+        "Box-Richards",
+        "Box-DeltaBlue",
+        "Box-RayTrace",
+        "Box-RegExp",
+        "Box-NavierStokes",
+        "Box-Crypto",
+        "Box-Splay",
+        "Box-SplayLatency",
+        "Box-EarleyBoyer",
+        "Box-PdfJS",
+        "Box-Mandreel",
+        "Box-MandreelLatency",
+        "Box-Gameboy",
+        "Box-CodeLoad",
+        "Box-Box2D",
+        "Box-zlib",
+        "Box-Typescript"
+    };
+
+    public void runUiAutomation() throws Exception {
+        final int timeout = 10;
+        Bundle status = new Bundle();
+        status.putString("product", getUiDevice().getProductName());
+
+        sleep(timeout);
+        UiSelector selector = new UiSelector();
+        /* Accept terms of service */
+        try {
+            UiObject acceptButton = new UiObject(selector.resourceId("org.chromium.chrome:id/lightweight_fre_terms_accept")
```

Comment:
Pushed changes. Still works for me and works on Nexus10 (with stock image) to the point where the octane2 is loaded. I've no idea how to resolve uiautomator dump problem with nexus10.